hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -326,6 +328,16 @@ private boolean onDeltaSyncShutdown(boolean error) {
     @Parameter(names = {"--checkpoint"}, description = "Resume Delta Streamer from this checkpoint.")
     public String checkpoint = null;
 
+    /**
+     * 1. string: topicName,partition number 0:offset value,partition number 1:offset value
+     * 2. timestamp: kafka offset timestamp
+     * example
+     * 1. hudi_topic,0:100,1:101,2:201
+     * 2. 1621947081
+     */
+    @Parameter(names = {"--checkpoint-type"}, description = "Checkpoint type, divided into timestamp or string offset")
+    public String checkpointType = "string";
```

Comment:
I am contemplating between "string" or "default" or "regular" to be set as default checkpoint type. @n3nash : any thoughts. We are looking to introduce a new config called checkpoint type. by default we need to set some value. this patch adds a new checkpoint type "timestamp" for kafka source.

Reply:
@nsivabalan Do we need to introduce something explicitly here ? Can we just introduce another property like below `hoodie.deltastreamer.source.kafka.checkpoint.type` and not have this change present as a top level option ? This checkpoint type seems very specific to a use-case in kafka and would like to reduce the confusions at the top level configs for users who want to use other source types.